components/engine_traits/src/raft_engine.rs

```diff
@@ -115,9 +115,16 @@ pub trait RaftEngine: RaftEngineReadOnly + PerfContextExt + Clone + Sync + Send
         Ok(total)
     }
 
+    fn need_manual_purge(&self) -> bool {
+        false
+    }
+
     /// Purge expired logs files and return a set of Raft group ids
     /// which needs to be compacted ASAP.
-    fn purge_expired_files(&self) -> Result<Vec<u64>>;
+    fn manual_purge(&self) -> Result<Vec<u64>> {
+        debug_assert!(false, "missing `manual_purge` implementation");
```

Comment:
Bad style. Better leave it unimplemented.

Reply:
The intention is to only panic in debug build, because this issue (call `manual_purge` when `need_manual_purge` returns false, or haven't implemented `manual_purge` when it returns true) is not a critical one.

Reply:
I understand the intention, but it's not a big deal to force the implementer to think whether it needs to do purge.